tests/pytests/test_vecsim.py

```diff
@@ -2394,3 +2390,28 @@ def test_switch_write_mode_multiple_indexes(env):
     if bg_indexing == 0:
         prefix = "::warning title=Bad scenario in test_vecsim:test_switch_write_mode_multiple_indexes::" if GHA else ''
         print(f"{prefix}All vectors were done reindex before switching back to in-place mode")
+
+
+def test_vector_index_ptr_valid(env):
+    conn = getConnectionByEnv(env)
+    # Scenerio1: Vecsim Index scheme with numeric (or non-vector type) and vector type with invalid parameter 
+    #            Insert partial doc - only numeric
+    #            Update Doc
+    
+    # HNSW parameters the causes an execution throw (M > UINT16_MAX)
+    UINT16_MAX = 2**16
+    M = UINT16_MAX + 1
+    dim = 2
+
+    res = conn.execute_command('FT.CREATE', 'idx','SCHEMA', 'n', 'NUMERIC',
+                    'v', 'VECTOR', 'HNSW', '8', 'TYPE', 'FLOAT16', 'DIM', dim, 'DISTANCE_METRIC', 'L2', 'M', M) 
+    env.assertEqual(res, 'OK')
+
+    res = conn.execute_command('HSET', 'doc', 'n', 0)
+    env.assertEqual(res, 1)
+    # Before bug fix, the following command would cause a server crash due to the null pointer accsess
+    res = conn.execute_command('HSET', 'doc', 'n', 1)
+    env.assertEqual(res, 0)
```

Comment:
Is there anything we can observe about the field (from the info/a debug command) that we can add to the test?

Reply:
Also, consider adding another command where we try to add a vector and validate the index error (to test the bad vector indexing creation on the way)

Reply:
> Is there anything we can observe about the field (from the info/a debug command) that we can add to the test?
We decided not to raise an error in this scenario